internal/ingress/controller/template/template.go

```diff
@@ -882,41 +884,87 @@ func proxySetHeader(loc interface{}) string {
 
 // buildCustomErrorDeps is a utility function returning a struct wrapper with
 // the data required to build the 'CUSTOM_ERRORS' template
-func buildCustomErrorDeps(proxySetHeaders map[string]string, errorCodes []int, enableMetrics bool) interface{} {
+func buildCustomErrorDeps(upstreamName string, errorCodes []int, enableMetrics bool) interface{} {
 	return struct {
-		ProxySetHeaders map[string]string
-		ErrorCodes      []int
-		EnableMetrics   bool
+		UpstreamName  string
+		ErrorCodes    []int
+		EnableMetrics bool
 	}{
-		ProxySetHeaders: proxySetHeaders,
-		ErrorCodes:      errorCodes,
-		EnableMetrics:   enableMetrics,
+		UpstreamName:  upstreamName,
+		ErrorCodes:    errorCodes,
+		EnableMetrics: enableMetrics,
 	}
 }
 
-// collectCustomErrorsPerServer is a utility function which will collect all
+// getDefaultBackendUpstreamNameForLocation returns the upstream-formatted name of the
+// default backend for a given location (either custom, specified by annotation, or
+// the global default)
+func getDefaultBackendUpstreamNameForLocation(input interface{}) string {
+	location, ok := input.(*ingress.Location)
+	if !ok {
+		klog.Errorf("expected a '*ingress.Location' type but %T was returned", input)
+		return ""
+	}
+
+	if location.DefaultBackend != nil {
+		return fmt.Sprintf("custom-default-backend-%v", location.DefaultBackend.GetName())
+	}
+	return "upstream-default-backend"
```

Comment:
I think these values are used in controller.go as well, how hard would it be to put them into a helper function and constant respectively and reuse here?

Reply:
I'd be happy to do that. package controller has a utils file already, but importing the controller package here doesn't quite seem right - is there somewhere with shared utils (across both controller and template)?

Reply:
Maybe we can make this function a property of `Location` struct?

I don't mind leaving this as is for now.

Reply:
I tried an implementation of making it a property of Location struct, let me know if it's what you had in mind?